src/large_pages/node_large_page.cc

```diff
@@ -282,20 +294,42 @@ bool IsSuperPagesEnabled() {
 }
 #endif
 
+// Functions in this class must always be inlined because they must end up in
+// the `lpstub` section rather than the `.text` section.
+class MemoryMapPointer {
+ public:
+  FORCE_INLINE MemoryMapPointer() {}
```

Comment:
Don't use `FORCE_INLINE` unless you've established beyond doubt that lack of inlining causes performance issues (which is exceedingly unlikely here.)

Reply:
@bnoordhuis It’s not about performance issues, see the comment above the function

Reply:
Okay, but I don't follow how `__attribute__((always_inline))` ensures it ends up in a non-standard section. Can you elaborate (and explain why `always_inline` is better than an explicit `__section__("lpstub")`)?

(I saw https://github.com/nodejs/node/pull/32570#discussion_r400837870 but the commit is gone and there's not enough context for me to reconstruct what the code looked like.)

Reply:
74779ed37f3c28d23db6bdffaea479c31ded80bc is the fixup commit that changed this. Always-inline works here because it’s inlined only into code that is already designated for the lpstub section. If we ever decide to reuse this class, putting its code into `lpstub` won’t make any sense, i.e. it makes the class a bit more self-contained.

It you feel strongly about this, it’s okay, I don’t and I suspect @gabrielschulhof doesn’t either.

Reply:
@bnoordhuis I'd like to keep `FORCE_INLINE` because the sole function currently present in `lpstub`, `MoveTextRegionToLargePages` is aligned at the 2 MB boundary. If we add more code, it will either end up right after this function, which would be great, but if it ends up in front of the function, it will not be aligned at the 2 MB boundary, so a gap in the section may appear. If we tell all functions to align at the 2 MB boundary, there will be many gaps in the `lpstub` section and it will be unnecessarily large.

That being said, I tried removing `FORCE_INLINE` and placing the class into the `lpstub` section as you asked and it turns out that the whole code gets inlined anyway because it's so simple. Thus, I would like to keep `FORCE_INLINE` in case some later compiler may decide to *not* inline the code.